File: src/main/java/lach_01298/moreBees/genetics/WitherSkullJubilance.java
package lach_01298.moreBees.genetics;

import forestry.api.apiculture.IAlleleBeeSpecies;
import forestry.api.apiculture.IBeeGenome;
import forestry.api.apiculture.IBeeHousing;
import forestry.api.apiculture.IJubilanceProvider;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.tileentity.TileEntitySkull;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;

import javax.annotation.Nonnull;

public class WitherSkullJubilance implements IJubilanceProvider {

	@Override
	public boolean isJubilant(@Nonnull IAlleleBeeSpecies species, @Nonnull IBeeGenome genome, @Nonnull IBeeHousing housing) {
		World world = housing.getWorldObj();
		BlockPos pos = housing.getCoordinates();

		TileEntity tile;
		do {
			pos = pos.up();
			tile = world.getTileEntity(pos);
		} while (tile instanceof IBeeHousing && pos.getY() > 0);

        return (tile instanceof TileEntitySkull && ((TileEntitySkull) tile).getSkullType() == 1);
	}

}

File: src/main/java/com/tencao/morebees/bees/WitherSkullJubilance.kt
package com.tencao.morebees.bees

import com.tencao.morebees.util.WorldHelper
import forestry.api.apiculture.IAlleleBeeSpecies
import forestry.api.apiculture.IBeeGenome
import forestry.api.apiculture.IBeeHousing
import forestry.api.apiculture.IJubilanceProvider
import net.minecraft.tileentity.TileEntity
import net.minecraft.tileentity.TileEntitySkull
import net.minecraft.util.math.AxisAlignedBB

class WitherSkullJubilance : IJubilanceProvider {

    /**
     * Why are you looking at this, you dirty cheater
     */
    override fun isJubilant(species: IAlleleBeeSpecies, genome: IBeeGenome, housing: IBeeHousing): Boolean {
        val world = housing.worldObj
        val pos = housing.coordinates

        val area = AxisAlignedBB(pos)
        area.grow(5.0, 2.0, 5.0)

        WorldHelper.getPositionsFromBox(area).forEach {
            val tile: TileEntity? = world.getTileEntity(it)
            if (tile is TileEntitySkull && tile.skullType == 1)
                return true
        }
        return false
    }

}